lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1731,6 +1882,61 @@ func (d *zfs) UnmountVolume(vol Volume, keepBlockDev bool, op *operations.Operat
 
 	refCount := vol.MountRefCountDecrement()
 
+	deactivate := func() (bool, error) {
```

Comment:
Moving this logic into a function makes sense. But by doing so you've lost the definition of the `ourUnmount` variable, leaving no explanation as to what the `bool` returned from this function is. Please can you add a comment above this line explaining what it does and what it returns.